certbot/display/util.py

```diff
@@ -197,18 +231,59 @@ def checklist(self, message, tags, default_status=True, **unused_kwargs):
             else:
                 return code, []
 
-    def directory_select(self, message, **unused_kwargs):
+    def _return_default(self, prompt, default, cli_flag, force_interactive):
+        """Should we return the default instead of prompting the user?
+
+        :param str prompt: prompt for the user
+        :param default: default answer to prompt
+        :param str cli_flag: command line option for setting an answer
+            to this question
+        :param bool force_interactive: if interactivity is forced by the
+            IDisplay call
+
+        :returns: True if we should return the default without prompting
+        :rtype: bool
+
+        """
+        msg = "Invalid IDisplay call for this prompt:\n{0}".format(prompt)
+        if cli_flag:
+            msg += ("\nYou can set an answer to "
+                    "this prompt with the {0} flag".format(cli_flag))
+        assert default is not None or force_interactive, msg
+        return not self._can_interact(force_interactive)
+
+    def _can_interact(self, force_interactive):
+        """Can we safely interact with the user?
+
+        :param bool force_interactive: if interactivity is forced by the
+            IDisplay call
+
+        :returns: True if the display can interact with the user
+        :rtype: bool
+
+        """
+        if self.force_interactive or force_interactive:
+            return True
+        else:
+            return sys.stdin.isatty() and self.outfile.isatty()
```

Comment:
When this happens, should we debug log that we're suppressing an interactive question because there's no TTY?